tests/pricing_models/test_calc_max_long_short.py

```diff
@@ -0,0 +1,190 @@
+
+    __test__ = False  # pytest: don't test this class
+
+
+class TestCalculateMax(unittest.TestCase):
+    """Tests calculate_max_short and calculate_max_long functions within the pricing model"""
+
+    def test_calculate_max_long(self):
+        """
+        Tests that calculate_max_long and calculate_max_short are safe, by checking
+            apr >= 0
+            share_price * market_state.share_reserves >= base_buffer
+            bond_reserves >= bond_buffer
+        """
+        log_utils.setup_logging(log_filename="test_calculate_max")
+        pricing_model: HyperdrivePricingModel = HyperdrivePricingModel()
+
+        # Values from Hyperdrive
+        # poolConfig
+        #     initialSharePrice 1000000000000000000
+        #     minimumShareReserves 1000000000000000000
+        #     positionDuration 31536000
+        #     checkpointDuration 86400
+        #     timeStretch 44463125629060298
+        #     governance 0x71554DE85ecD7bDD19e24078e518ead88d691871
+        #     feeCollector 0xd002315CAcB4882e1099EDFb00895Fa8867256B5
+        #     fees.curve 0
+        #     fees.flat 0
+        #     fees.governance 0
+        #     oracleSize 5
+        #     updateGap 1000
+
+        # poolInfo
+        #     shareReserves 500000000000000000000000000
+        #     bondReserves 1498059016940075710500000000
+        #     lpTotalSupply 499999999000000000000000000
+        #     sharePrice 1000000000000000000
+        #     longsOutstanding 0
+        #     longAverageMaturityTime 0
+        #     shortsOutstanding 0
+        #     shortAverageMaturityTime 0
+        #     shortBaseVolume 0
+        #     withdrawalSharesReadyToWithdraw 0
+
+        # baseAmount 493213221042049515844300901
+        # bondAmount 504845795898026194655699099
+
+        # poolInfo
+        #     shareReserves 993213221042049515844300901
+        #     bondReserves 993213221042049549613550417
+        #     lpTotalSupply 499999999000000000000000000
+        #     sharePrice 1000000000000000000
+        #     longsOutstanding 504845795898026160886449583
+        #     longAverageMaturityTime 126144000000000000000000000
+        #     shortsOutstanding 0
+        #     shortAverageMaturityTime 0
+        #     shortBaseVolume 0
+        #     withdrawalSharesReadyToWithdraw 0
+        #     withdrawalSharesProceeds 0
+
+        test_case = TestCaseCalcMax(
+            market_state=HyperdriveMarketState(
+                share_reserves=FixedPoint(scaled_value=500000000000000000000000000),
+                bond_reserves=FixedPoint(scaled_value=1498059016940075710500000000),
+                lp_total_supply=FixedPoint(scaled_value=499999999000000000000000000),
+                init_share_price=FixedPoint(1),
+                share_price=FixedPoint(1),
+                curve_fee_multiple=FixedPoint(0),
+                flat_fee_multiple=FixedPoint(0),
+            ),
+            time_remaining=time.StretchedTime(
+                days=FixedPoint("90"),
```

Comment:
up to you how you want to construct this, but just FYI you can construct with `int` now and it'll scale it. i.e. `FixedPoint("90") == FixedPoint("90.0") == FixedPoint(90) == FixedPoint(90.0)`